Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -1,25 +1,74 @@
+import React, { useState, FormEvent } from "react";
 import BatteryInput from "~/components/batteryInput";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
-  }
+  const [speed, setSpeed] = useState(NaN);
+  const [battery, setBattery] = useState(NaN);
+  const [weather, setWeather] = useState(0);
+  const [range, setRange] = useState(NaN);
+  const [speedError, setSpeedError] = useState("");
+  const [batteryError, setBatteryError] = useState("");
```

Comment:
Do we need to use state here?

Reply:
Think about it this way, if this teaser had 50 input fields, would it be feasible to create 50 state variables to represent the error for each one?